File: nl.naturalis.nba.client.php/model/Sex.php
<?php

namespace nl\naturalis\nba\client\php\model;

class Sex {

	private static $male;
	private static $female;
	private static $mixed;
	private static $hermaphrodite;

	public static function MALE()
	{
		if (self::$male === null)
			self::$male = new Sex("male");
		return self::$male;
	}

	public static function FEMALE()
	{
		if (self::$female === null)
			self::$female = new Sex("female");
		return self::$female;
	}

	public static function HERMAPHRODITE()
	{
		if (self::$hermaphrodite === null)
			self::$hermaphrodite = new Sex("hermaphrodite");
		return self::$hermaphrodite;
	}

	private final $name;

	private function __construct(string $name)
	{
		$this->name = $name;
	}

	public function __toString()
	{
		return $this->name;
	}
}

File: nl.naturalis.nba.api/src/main/java/nl/naturalis/nba/api/model/Sex.java
package nl.naturalis.nba.api.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

public enum Sex implements INbaModelObject
{

	MALE, FEMALE, MIXED, HERMAPHRODITE;

	@JsonCreator
	public static Sex parse(@JsonProperty("name") String name)
	{
		if (name == null) {
			return null;
		}
		for (Sex sex : Sex.values()) {
			if (sex.name.equalsIgnoreCase(name)) {
				return sex;
			}
		}
		throw new IllegalArgumentException("Invalid sex: " + name);
	}

	private final String name = name().toLowerCase();

	@JsonValue
	@Override
	public String toString()
	{
		return name;
	}

}
